### PR TITLE
Treat optional Set params default to Array

### PR DESCRIPTION
**Pull request by u2**
Is it reasonable for adding `Set` nested params?

### CODE REVIEW

File: spec/grape/validations/params_scope_spec.rb [hunk 1]
```diff
@@ -227,7 +227,7 @@ def app
     it 'errors with an unsupported type' do
       expect do
         subject.params do
-          group :a, type: Set do
+          group :a, type: Rational do
```
Comment (dblock) [inline spec/grape/validations/params_scope_spec.rb]: Could you please explain this one?

**Comment by dblock** [comment]
I would say "yes' if with 2 parameters with identical values you would get an error (or you would get only 1 value). Otherwise this has no chance of behaving any different than an array, no?

File: spec/grape/validations/params_scope_spec.rb (hunk 1 above)
Comment (u2) [inline spec/grape/validations/params_scope_spec.rb]: Sorry, I was napping at that moment.

**Comment by u2** [comment]
Yes.I think it's better to get only 1 value.And I think we need more specs for going on.

**Comment by dblock** [comment]
Bump @u2.

**Comment by u2** [comment]
Mm hmm, I nearly forgot this.I will do it later.But it may be not a easy work.So could you merge it firstly,and I will open another PR for further work.

**Comment by dblock** [comment]
Lets add full Set support so that we don't have to change behavior for clients in the future. I'll leave this open, maybe @u2 or someone else can finish this?

**Comment by u2** [comment]
Sorry about leaving aside the problem until now.
I reconsider about whether it's necessary for us and its rationality.
Firstly, for the params `Array` and `Set` are the same expression for http params.
Secondly, I have search issues about `Set` and I could not find anybody cared about that.

> Set implements a collection of unordered values with no duplicates. This is a hybrid of Array's intuitive inter-operation facilities and Hash's fast lookup.

In ruby, Set is different with `Array`.But, for us, it's just a uniq `Array`,and if it was a `Set`, what about the meaning of `unordered values`?

Before I committed this PR about `Set`, just because I saw something like
https://github.com/ruby-grape/grape/blob/d721c5d7643eea6f2ecfc92ef9dc4c529daf9762/lib/grape/validations/types.rb#L54.

What's your opinion? @dblock

**Comment by dblock** [comment]
I think it could be valuable, a Set is a fairly basic structure. But clearly not many people care :) I'll just leave this open, feel free to finish it.

**Pull request closed without merging**